extensions/resource-deployment/src/test/azdataService.test.ts

```diff
@@ -0,0 +1,93 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'mocha';
+import * as TypeMoq from 'typemoq';
+import assert = require('assert');
+import { IPlatformService, CommandOptions } from '../services/platformService';
+import { AzdataService } from '../services/azdataService';
+import { BdcDeploymentType } from '../interfaces';
+
+suite('azdata service Tests', function (): void {
+	test('azdata service handles deployment types properly', async () => {
+		const mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();
+		const azdataService = new AzdataService(mockPlatformService.object);
+		mockPlatformService.setup((service) => service.runCommand(TypeMoq.It.isAnyString(), TypeMoq.It.isAny())).returns((command: string, options: CommandOptions | undefined) => {
+			return new Promise<string>((resolve) => {
+				resolve('{"result":[]}');
+			});
+		});
+
+		azdataService.getDeploymentProfiles(BdcDeploymentType.ExistingAKS);
+		azdataService.getDeploymentProfiles(BdcDeploymentType.ExistingARO);
+		azdataService.getDeploymentProfiles(BdcDeploymentType.ExistingKubeAdm);
+		azdataService.getDeploymentProfiles(BdcDeploymentType.ExistingOpenShift);
+		azdataService.getDeploymentProfiles(BdcDeploymentType.NewAKS);
+
+		try {
+			azdataService.getDeploymentProfiles(<BdcDeploymentType>'no-such-type');
```

Comment:
For example here you could just do

`should(azdataService.getDeploymentProfiles(...)).throw();`

Reply:
i also loke should better. However, if you wanted to stick with assert, assert.throws can be used for equvilent way of the should statement that Charlse has suggested above.